cardano-testnet/src/Testnet/Cardano.hs

```diff
@@ -700,15 +654,17 @@ cardanoTestnet testnetOptions H.Conf {..} = do
   H.assertIO . IO.doesFileExist $ tempAbsPath </> "tx1.tx"
 
   -- Add Byron, Shelley and Alonzo genesis hashes to node configuration
+  -- TODO: These genesis filepaths should not be hardcoded. Using the cli as a library
+  -- rather as an executable will allow us to get the genesis files paths in a more
+  -- direct fashion.
```

Comment:
If we test using the library rather than the executable, it would mean that we aren't testing the CLI parsing code.  Is this definitely the direction we want to go?

If we do head down this way, there is something we need to watch out for.  Some CLI does error handling with `error` which doesn't give us a clear interface with which to catch errors and handle them.  We should probably be more rigorous with how we error in the command implementation and not use `error` if we go down this way so tests can handle those cases.

Reply:
> If we test using the library rather than the executable, it would mean that we aren't testing the CLI parsing code. Is this definitely the direction we want to go?

I think so. We can separately test individual commands to see if they parse correctly but its not needed to launch a testnet and should simplify things.

Reply:
> Some CLI does error handling with error which doesn't give us a clear interface with which to catch errors and handle them. We should probably be more rigorous with how we error in the command implementation and not use error if we go down this way so tests can handle those cases.

Good point :+1: